airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/dest_state_lifecycle_manager/DestStateLifecycleManager.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.dest_state_lifecycle_manager;
+
+import io.airbyte.protocol.models.AirbyteMessage;
+import java.util.Queue;
+
+/**
+ * This class manages the lifecycle of state message. It tracks state messages that are in 3 states:
+ * <ol>
+ * <li>pending - associated records have been accepted by the connector but has NOT been pushed to
+ * the destination</li>
+ * <li>flushed - associated records have been flushed to tmp storage in the destination but have NOT
+ * been committed</li>
+ * <li>committed - associated records have been committed</li>
+ * </ol>
+ */
+public interface DestStateLifecycleManager {
+
+  /**
+   * Accepts a state into the manager. The state starts in a pending state.
+   *
+   * @param message - airbyte message of type state
+   */
+  void addState(AirbyteMessage message);
+
+  /**
+   * Moves any tracked state messages that are currently pending to flushed.
+   */
+  void markPendingAsFlushed();
+
+  /**
+   * List all tracked state messages that are flushed.
+   *
+   * @return list of state messages
+   */
+  Queue<AirbyteMessage> listFlushed();
```

Comment:
Nitpick. In test, it is nice to have this `listFlushed` method to inspect the elements. However, in production, we only need to know whether there is any flushed state or not. What about adding a `hasFlushed` method specifically for production to eliminate the overhead of creating those lists over and over?